code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/GithubExternalIdHandler.java

```diff
@@ -77,7 +77,15 @@ public ExternalId untransform(ExternalId externalId) {
         if(StringUtils.isBlank(jwt) && !StringUtils.isBlank(accessToken)){
             try {
                 jwt =  ProjectConstants.AUTH_TYPE + githubTokenHandler.getGithubToken(accessToken).getJwt();
-            } catch (IOException e) {
+            } catch (ClientVisibleException e) {
```

Comment:
I don't think this actually deletes the object from the database.  But it doesn't seem to be the right approach to modify the account during a call like this.  It can really cause things to get out of sync.  What's the driving motivation for this change?